### ISSUE TITLE
Call for New Games

### ISSUE THREAD

**Issue by lanctot** [google-deepmind/open_spiel#843]
Hi all, 

I'm starting a thread to collect requests for new games and plans for upcoming games. We will keep this updated as new games get released, and as requests come in (please feel free to comment on this thread to add a request). 

Games we would like to add to OpenSpiel:

- Ant foraging on a grid
- Capture-the-Flag on a gridworld (or other team adversarial games)
- [Chinese Checkers](https://en.wikipedia.org/wiki/Chinese_checkers)
- [Chinese Chess](https://en.wikipedia.org/wiki/Xiangqi) (see https://github.com/deepmind/open_spiel/issues/802)
- [Chinese Dark Chess](https://en.wikipedia.org/wiki/Banqi)
- [Mahjong](https://en.wikipedia.org/wiki/Mahjong)
- Pursuit-evasion game played on a graph or grid
- [Shogi](https://en.wikipedia.org/wiki/Shogi)

Implementing one of these games is a great way to get involved with contributing to OpenSpiel. If you would like to volunteer to implement one, please contact us and let us know as we might already be working on them. 

Games that have been added since posting this thread:
- [2048](https://en.wikipedia.org/wiki/2048_(video_game)) (Thanks @Jazeem. PR now merged: https://github.com/deepmind/open_spiel/pull/894)
- Checkers (thanks @Jazeem! PR now merged: https://github.com/deepmind/open_spiel/pull/861)
- Euchre (thanks @jhtschultz! Added in https://github.com/deepmind/open_spiel/commit/a3dbb1b7483b3b02077a92bb3a84389907afb79b)
- Mancala (Kalah) (thanks @Jazeem! see https://github.com/deepmind/open_spiel/pull/841)
- [Nim](https://en.wikipedia.org/wiki/Nim) (thanks @acforvs! PR merged: https://github.com/deepmind/open_spiel/pull/868)
- Phantom Go (thanks @Syor! PR now merged: https://github.com/deepmind/open_spiel/pull/867)
- [Dou dizhu](https://en.wikipedia.org/wiki/Dou_dizhu): (Thanks @rezunli96! PR merged: https://github.com/deepmind/open_spiel/pull/969)
- [Mensch-Ärgere-Dich-Nicht](https://en.wikipedia.org/wiki/Mensch_%C3%A4rgere_Dich_nicht) (Thanks @thorsten-j ! PR merged: https://github.com/deepmind/open_spiel/pull/915)
- Dominoes (Thanks @morLev ! PR merged: https://github.com/deepmind/open_spiel/pull/1018)
- [Nine Men's Morris](https://en.wikipedia.org/wiki/Nine_men%27s_morris). PR merged: https://github.com/deepmind/open_spiel/pull/1079
- [EinStein würfelt nicht!](https://en.wikipedia.org/wiki/EinStein_w%C3%BCrfelt_nicht!) (Thanks @giogix2! PR merged: https://github.com/google-deepmind/open_spiel/pull/1289)
- [Cribbage](https://en.wikipedia.org/wiki/Cribbage). Done! Added in https://github.com/google-deepmind/open_spiel/commit/82819d0121744531affb6dffa4ea52b159ab8066
- Latent Tic-Tac-Toe (see description in [this paper](http://mlanctot.info/files/papers/nips09mccfr.pdf)) Done! Added in https://github.com/google-deepmind/open_spiel/pull/1361. Thanks @giogix2!
- [Yacht](https://en.wikipedia.org/wiki/Yacht_(dice_game)) (Thanks @gmweinberg! PR merged: https://github.com/google-deepmind/open_spiel/pull/1403)

With these additions, OpenSpiel now has 95 games.

**Comment by Jazeem** [comment]
- Mahjong
- Reversi
- Yacht
- Dominoes

Could be good additions

**Comment by lanctot** [comment]
We already have Othello (Reversi), but I will add the others, thanks!

**Comment by acforvs** [comment]
What are your thoughts on adding 2048? 
The original repo https://github.com/gabrielecirulli/2048 is under an MIT license, so I hope there shouldn't be any problems regarding copyright..

**Comment by lanctot** [comment]
> What are your thoughts on adding 2048? The original repo https://github.com/gabrielecirulli/2048 is under an MIT license, so I hope there shouldn't be any problems regarding copyright..

Great idea, I'll add it to the list!

**Comment by dipplestix** [comment]
Is anyone working on Cribbage? I'd be willing to take on that one

**Comment by lanctot** [comment]
> Is anyone working on Cribbage? I'd be willing to take on that one

I would really, really _love_ to see Cribbage in the code base! I had some interactions with a redditor who said they would look into it, but have not heard from them in a while. I'll touch base with them and point them to this conversation and let you know soon.

**Comment by lanctot** [comment]
@dipplestix Ok heard back from them: they got replied saying they had been busy and was thinking of getting back to it in the fall, but it shouldn't stop someone else. If you finish it before the summer ends then maybe they could even help (they are thinking of doing research in the game). So, it sounds like you should go for it :) If you'd like to contact them, contact me by email and can give their reddit contact.

**Comment by PrashantPD** [comment]
Hi, I would like to work on the Pursuit-Evasion game played by 2 competitors say for instance: a company that employs an outside consultant and is receptive to recommendations and a company that makes decisions in-house as a canonical model deploying game theory. I will use the multi-armed bandit algorithm in which the company that employs an outside consultant will receive pricing recommendations based on historical moves to facilitate its decision making whereas the company that primarily takes decisions in-house can decide on the probability distribution for the choices available to it and the corresponding value attached to such a distribution.

**Comment by lanctot** [comment]
Hi @PrashantPD, ok I'm not seeing the actual game itself. Is there an example you can point to in e.g. a game theory text or Wikipedia? We're always welcoming of new games. It always helps if they are well-known in the MARL or game theory literature, but they are welcome either way (for now though, I'm curious about the structure of the game in extensive-game format). However, maybe let's continue this on the other thread you opened. I'll follow-up there.

**Comment by PrashantPD** [comment]
> Hi @PrashantPD, ok I'm not seeing the actual game itself. Is there an example you can point to in e.g. a game theory text or Wikipedia? We're always welcoming of new games. It always helps if they are well-known in the MARL or game theory literature, but they are welcome either way (for now though, I'm curious about the structure of the game in extensive-game format). However, maybe let's continue this on the other thread you opened. I'll follow-up there.

Hi @lanctot, Absolutely - thanks see you there - will be on it and detail the structure of the game and point to the relevant literature. You should see it first thing tomorrow in the morning as you log in. The multi-armed bandit comprising of a player and the environment which can comprise of the competition struck a cord in a pursuit-evasion context where the competitive game ends in an equilibrium.

**Comment by PrashantPD** [comment]
Hi @lanctot I am working on the game for the other thread today - will message tomorrow on both the formats.

**Comment by lanctot** [comment]
> Hi @lanctot I am working on the game for the other thread today - will message tomorrow on both the formats.

Thanks for letting me know, but there is no rush. And I may not replay for a day or two depending on how busy I am.

**Comment by thorsten-j** [comment]
Hi Marc,

on Gravon, Mahjong already is implemented. I could try to port the code 
to C++/OpenSpiel, if nobody else volunteers.

Best regards,
Thorsten

**Comment by lanctot** [comment]
> Hi Marc, on Gravon, Mahjong already is implemented. I could try to port the code to C++/OpenSpiel, if nobody else volunteers. Best regards, Thorsten

Cool! 

@rezunli96 suggested that we was going to look into it after hus last PR which we just merged a few days ago; Zun, have you already started?

@thorsten-j: do you know if there would be any license incompatibilities?

**Comment by rezunli96** [comment]
Hi @thorsten-j do you know which version of Mahjong Gravon has? The one I am currently looking into is Riichi Mahjong.

**Comment by thorsten-j** [comment]
Hi @rezunli96 , Gravon implements Hong Kong Mahjong.
Hi @lanctot , afaik there is no license needed for Mahjong.

**Comment by rezunli96** [comment]
@thorsten-j Sounds great! Then I think we do not really have a conflict here and we can work on our own versions of Majong separately. Maybe there is a unified implementation that can include all versions. But I guess this is not the primary concern for now?

**Comment by lanctot** [comment]
> Hi @lanctot , afaik there is no license needed for Mahjong.

Right, that should be fine .. what I meant is, is the code on Gravon have a license like e.g. GLPv3? (Oh, maybe the Gravon code is not open sourced? Is it your own personal implementation?)

**Comment by thorsten-j** [comment]
Gravon's code is not open source but it's written by myself.

**Comment by lanctot** [comment]
@dipplestix I am thinking of implementing Cribbage (as a slow side project). Just wondering if you plan to finish/contribute yours?

**Comment by lanctot** [comment]
> Gravon's code is not open source but it's written by myself.

Hong Kong Mahjong would be a great addition. And it seems like @rezunli96 is working on a different game anyway. Does it have any "interruptions" as mentioned in https://github.com/deepmind/open_spiel/issues/979 ? If so, I'm curious: how did you deal with them in a discrete turn-based API?

**Comment by thorsten-j** [comment]
> Does it have any "interruptions" as mentioned in #979 ? If so, I'm curious: how did you deal with them in a discrete turn-based API?

On Gravon, after a player discarded a tile, all other players have to either call that tile or pass. This way Gravon makes sure that no one misses a tile. For implementation that means that there is no interrupt but the game still is completely turn based.

**Comment by TheSQLGuru** [comment]
Are there any war gamers, especially hex-grid style with dice or some other randomization factor such as card draw/play, to be found here? I reviewed the list of games and didn't notice any similar to the types I mention. I am just getting started in the AI/ML realm for a variety of reasons (after decades of consulting the SQL Server database applications realm) and have set myself a Bucket List item to build an AI opponent for the Advanced Squad Leader Starter Kit game. Having a (semi-)similar game already done in Open Spiel would obviously GREATLY benefit my pursuit! :-D

It is amazing that this project exists, and even more amazing is the number of games and algorithms that have been implemented in just 3 years!! I truly look forward to the time when I can understand and dig into the how's and why's of all that Open Spiel has to offer!

**Comment by lanctot** [comment]
Hi @TheSQLGuru  

>  (after decades of consulting the SQL Server database applications realm) 

Username checks out! :-p 

OpenSpiel does not have any games in this category. It'd be a great addition to the library, but they are fairly complex to implement and sometimes it can be hard to include due to copyrights if it's a commercial game. But if there are any smaller ones that are non-commercial that can be used as an example, it'd be great to have in the repos. I think there are some good ideas of games that could fit this, and happy to brainstorm here or in a separate thread.

**Comment by TheSQLGuru** [comment]
Hello @lanctot - thanks for the quick reply! HAH, funny that you checked me out. I don't have much out there on the intertubz because I don't blog, stopped using twitter years ago, have had some major health issues since 2017 that drastically cut both my work volume and my travel to present at conferences around the world. Actually, I don't even have a website for my business, nor even a resume/CV, and I have been consulting since the mid-90's and started my company in 1999. But I do have several C-level references that will state that I saved/made their company. :-D 

On to business: Let's start another thread somewhere/how. I will leave that to you - just let me know where to show up. I will collect some basic information from you (and from some of this project's files), especially concerning the various types of information, metadata, states and transitions, mechanics/decisions/rules/etc. and the potential ease/difficulty with which each element may be translated to Open Spiel. I will hit up various gaming forums and ask for inputs about the simplest games of the genre. I will review them and pick a few, and then distill enough info from them that you and I can have an informed decision about which might be the easiest/best to pursue first.

**Comment by lanctot** [comment]
> I'd like to extend/add the most popular version of dominoes in Latin America, **played with 4 players instead of two**. There's already a great library under MIT license (see [Dominoes library](https://dominoes.readthedocs.io/en/latest/#game)). I'd be interested in adapting it for OpenSpiel in Python and potentially also in C++.
> 
> Please let me know if that's okay, and point me to some useful resources if you can :) I've been trying to follow the guide in the docs but I haven't fade much headwind yet, I guess its normal since I'm not used to working on projects of this size.

That would be great! I assume you mean as a python game which would wrap that Dominoes library.

I know the project has gotten quite big and can be intimidating, but we are happy to help you with this. Probably the best place to start is to follow-up on the other thread you started (https://github.com/google-deepmind/open_spiel/issues/1204) where I asked a few questions that need to be answered and also gave general advice. 

Since most of the games are in C++, how to add new Python game is not overly well-documented, and we will add extra bits of missing documentation as we go along.

**Comment by i-Madsen** [comment]
Hi @lanctot , I'm trying to take a stab at implementing Spades.

Since Bridge is somewhat similar, I'm using its implementation as a starting point, but may periodically have some questions/need some help. The game rules and such are easy enough to change, but I'm still pretty new to RL related stuff.

Is it ok to make a help thread for making Spades here?

**Comment by lanctot** [comment]
I would absolutely love to see Spades in OpenSpiel! Yes, by all means feel free to start a thread for Spades. If you are new to OpenSpiel, the best place to start would be the video tutorial to get familiar with the API: https://youtu.be/8NCPqtPwlFQ?si=GiFg3x6P9Lm6McGX

**Comment by jin404861445lan** [comment]
Hello, I am from China and want to implement the  [Chinese Dark Chess](https://en.wikipedia.org/wiki/Banqi)
 game. Have you already studied this game?
Is this game type Non-deterministic and with Imperfect information? Can I refer to Kriegspiel for implementation? If I can implement this game, I am willing to contribute the source code and look forward to your answer.

**Comment by jin404861445lan** [comment]
> already be working o

Hello, I want to implement the  [Chinese Dark Chess](https://en.wikipedia.org/wiki/Banqi) game. 
Have you already studied this game?
Is this game type Non-deterministic and with Imperfect information?
Can I refer to Kriegspiel for implementation? 
If I can implement this game, I am willing to contribute the source code and look forward to your answer.

**Comment by jin404861445lan** [comment]
Hello, I want to implement the  [Chinese Dark Chess](https://en.wikipedia.org/wiki/Banqi) game. 
Have you already working on this game?
Is this game type Non-deterministic and with Imperfect information?
Can I refer to Kriegspiel for implementation? 
If I can implement this game, I am willing to contribute the source code and look forward to your answer.

**Comment by lanctot** [comment]
Hi @jin404861445lan, 

We are not working on this game. It is easier to think of this game as a perfect information game with chance events (where every time a piece is revealed, a chance event determines which one it is). This would make it easier for search methods like MCTS and expectiminimax to play this games. Something like backgammon should be a good starting point. Kriegspiel probably uses the underlying chess implementation which you would not want to do -- you'd want to write it from scratch.

**Comment by shinoi2** [comment]
Hello, has anyone implemented [Gomoku](https://en.wikipedia.org/wiki/Gomoku)? If not I'd like to help get it done.

**Comment by lanctot** [comment]
> Hello, has anyone implemented [Gomoku](https://en.wikipedia.org/wiki/Gomoku)? If not I'd like to help get it done.

No, that would be a welcome addition!

**Comment by AussieSeaweed** [comment]
I think a better addition than gomoku is an m,n,k-game (generalization of gomoku). @shinoi2 Are you still working on it? If not, I can contribute.

**Comment by AussieSeaweed** [comment]
Hi, @lanctot, what are your thoughts about a Python game wrapper of [PokerKit](https://github.com/uoftcprg/pokerkit) for non-holdem poker variants? I am thinking of doing it.

**Comment by dmarx** [comment]
[*The Resistance*](https://en.wikipedia.org/wiki/The_Resistance_(game)) (original version, no plot cards or inquisitor role). The Resistance is a social deduction game, similar to Mafia/Werewolf. The game is extremely simple:

* a minority of players are randomly assigned to an opposing team. The opposing team members have full knowledge of everyone's role, the players on the majority team do not.
* at the beginning of each round, players share information (publicly) to try to influence the outcome of the round.
* the information sharing culminates in a public vote to choose a subset of the group to comprise a "team" for that round.
* once a team is successfully chosen, the team members have a blind vote to determine the outcome of the mission.
* the goal of the minority team is to "sabotage" some fraction of the missions. If they achieve this, the minority team wins. Otherwise, the majority team wins.

**Comment by lanctot** [comment]
> [_The Resistance_](https://en.wikipedia.org/wiki/The_Resistance_(game)) (original version, no plot cards or inquisitor role). The Resistance is a social deduction game, similar to Mafia/Werewolf. The game is extremely simple:
> 
>     * a minority of players are randomly assigned to an opposing team. The opposing team members have full knowledge of everyone's role, the players on the majority team do not.
> 
>     * at the beginning of each round, players share information (publicly) to try to influence the outcome of the round.
> 
>     * one player gets to nominate a "team" of players, who will determine the outcome of the round.
> 
>     * the information sharing culminates in a public vote. if the vote fails, the next player in the sequence gets to nominate the team for the round.
> 
>     * once a team is successfully chosen, the team members have a blind vote to determine the outcome of the round's "mission". a "mission" is only successful if the vote is unanimous.
> 
>     * the goal of the minority team is to "sabotage" some fraction of the missions. If they achieve this, the minority team wins. Otherwise, the majority team wins.
> 
> 
> A minimal implementation could do the pre-vote information sharing as just communicating a credible interval for what a player wants the rest of the group to think their beliefs are wrt the other players' roles.

@dmarx, The Resistence is an excellent game indeed, but to add something that is copyrighted is difficult (see https://github.com/google-deepmind/open_spiel/issues/770). We'd ultimately want permission to host an open implementation in our repos, which is not impossible but sometimes difficult and a long process. Alternatively it can be hosted in an external repos and we can list it as a third-party (external) game on our games list.

**Comment by shinoi2** [comment]
> I think a better addition than gomoku is an m,n,k-game (generalization of gomoku). @shinoi2 Are you still working on it? If not, I can contribute.

My work is here https://github.com/shinoi2/open_spiel/tree/master/open_spiel/games/renju
But it has some code style issues and hard code.

**Comment by lanctot** [comment]
> Is anyone working on Cribbage? I'd be willing to take on that one

Cribbage has now been added! https://github.com/google-deepmind/open_spiel/commit/82819d0121744531affb6dffa4ea52b159ab8066

**Comment by kurtcz** [comment]
Mariáš is a great card game for 3 people that could be added. One player plays against two opponents who play together.
Although I am new to OpenSpiel I would be excited to help with that.

**Comment by edwardchalstrey1** [comment]
Is there a guide somewhere on how to add a game?

**Edit:** guide here: https://openspiel.readthedocs.io/en/latest/developer_guide.html#adding-a-game

**Comment by Spartan-X1** [comment]
I’m interested in contributing a **gridworld-based Capture-the-Flag** game to OpenSpiel as part of this issue. Before I begin, I’d like to clarify a few points:

### 🔹 Clarifications Needed

1. **API / Design Expectations**  
   - Are there any preferred API patterns, abstractions, or templates I should follow?  
   - Is there an existing game in OpenSpiel that would be best to base this on?

2. **Game Rules & Constraints**  
   - Do you have any preferences regarding:
     - Turn-based vs. simultaneous moves?
     - Grid size or topology?
     - Full vs. partial observability?
     - Deterministic vs. stochastic dynamics?

3. **Reference Implementation**  
   - Is there any reference game or CTF variant (e.g., something similar to *BombSquad CTF* or another benchmark) that you would like this to resemble?

**Comment by lanctot** [comment]
Hi @Spartan-X1 , that sounds great!

You can take a look at laser_tag, which is probably the closest. Cooperative path-finding and box-pushing might be others. For the game rules, slightly prefer simultaneous to keep consistent with games in this category. The rest is mostly up to you ! Choose the version you like best.. it's your contribution, after all.  

No strong opinions on abstractions or API, just please try to stick closely to the general style of the other games (see e.g. laser_tag as an example). 

For reference implementations, the one from MeltingPot might make sense for inspiration: https://arxiv.org/pdf/2211.13746 but there are zero requirements to stay consistent with them on our side (if you want to resemble e.g. BombSquad CTF that'd also be fine). 

Make it your own! As long as the main dynamics of CTF are captured I think it'd make a good addition. I think the main thing is that OpenSpiel doesn't have many games of that type (adversarial team games on a grid) so it'd be great to have one, even if it's a pretty basic version.

**Comment by Spartan-X1** [comment]
Sure will get to it 😸

**Comment by Spartan-X1** [comment]
Hello @lanctot  , this issue is closed does this mean the CTF game is implemented or that is still open for contribution ?

**Comment by lanctot** [comment]
> Hello [@lanctot](https://github.com/lanctot) , this issue is closed does this mean the CTF game is implemented or that is still open for contribution ?

Nope! That was a mistake, this one should remain open.

**Comment by gmweinberg** [comment]
I am ready to start work on shogi.

**Comment by lanctot** [comment]
> I am ready to start work on shogi.

Go for it! Thanks for all the games. Your contributions will make a nontrivial part of the OpenSpiel 2.0 change log! 😁

**Comment by ClauYUN** [comment]
Is there anyone who is interested in the ban&pick drafting strategy of League of Legends?

**Comment by fumin** [comment]
I wonder if it is *legal* for open_spiel to accept an implementation of the game [四国军棋](https://zh.wikipedia.org/zh-cn/%E5%9B%9B%E5%9B%BD%E5%86%9B%E6%A3%8B) (herein tentatively translated as Four-state-war)?

Although, Four-state-war shares the same [ancestor](https://en.wikipedia.org/wiki/L%27Attaque) as Stratego, I believe it is not copyrighted, thanks to its also not copyrighted [parent](https://en.wikipedia.org/wiki/Luzhanqi).

Four-state-war is a two team, four player game. Differences between Four-state-war and Stratego include railroads (for long distance transport, similar to Chinese Chess chariots), sappers (akin to special forces on railroads), camps (special regions belonging to the flag, similar to Chinese Chess camps), bunkers (places that are safe from attacks).

**Comment by lanctot** [comment]
> I wonder if it is _legal_ for open_spiel to accept an implementation of the game [四国军棋](https://zh.wikipedia.org/zh-cn/%E5%9B%9B%E5%9B%BD%E5%86%9B%E6%A3%8B) (herein tentatively translated as Four-state-war)?
> 
> Although, Four-state-war shares the same [ancestor](https://en.wikipedia.org/wiki/L%27Attaque) as Stratego, I believe it is not copyrighted, thanks to its also not copyrighted [parent](https://en.wikipedia.org/wiki/Luzhanqi).
> 
> Four-state-war is a two team, four player game. Differences between Four-state-war and Stratego include railroads (for long distance transport, similar to Chinese Chess chariots), sappers (akin to special forces on railroads), camps (special regions belonging to the flag, similar to Chinese Chess camps), bunkers (places that are safe from attacks).

Four state war and Luzhanqi should be no problem if they are not copyrighted. Happy to confirm with our legal team if/when someone would offer to implement it.  I love the idea of a team-based version of stratego with railroads and camps! <3

**Comment by TheSQLGuru** [comment]
I have been out of pocket for, well, a very long time. Do we have a game yet that uses a hex-based map, with pieces that move about on this type of grid, with varying things like terrain, a variety of movement types and rules, maybe Line of Sight, adjacent and ranged interactions, etc., etc? I have a game I would very much like to create here, but I don't have the chops at all to create the infrastructure like the above that would be required for the implementation.

My apologies if this is not the place for this post.

**Comment by lanctot** [comment]
> I have been out of pocket for, well, a very long time. Do we have a game yet that uses a hex-based map, with pieces that move about on this type of grid, with varying things like terrain, a variety of movement types and rules, maybe Line of Sight, adjacent and ranged interactions, etc., etc? I have a game I would very much like to create here, but I don't have the chops at all to create the infrastructure like the above that would be required for the implementation.

We do not have anything like this.. closest thing is probably Laser Tag, but it could certainly accomodate something like this (sounds like a D&D combat like game?)

**Comment by kvr06-ai** [comment]
This adds Chinese Checkers from the wish list. 121-position star board, 2-6 players, with chain hop mechanics and undo support. All C++ tests pass (random sim, undo consistency, multi-player configs, hop chains, win condition). PR: #1513

**Comment by kvr06-ai** [comment]
Opened #1544 to add Capture-the-Flag (1v1, simultaneous moves) from the wish list.

**Comment by MahdiHoseinpoor** [comment]
I'd like to implement Shelem - a popular classic Persian 4-player partnership trick-taking card game - in OpenSpiel.

The game combines a bidding phase with imperfect-information trick-taking gameplay, which could make it an interesting environment for reinforcement learning research.

The game is played between two fixed teams (0 & 2 vs 1 & 3), where players only observe their own hand, the bidding history, and previously played cards. 

I plan to implement it in C++ under `open_spiel/games/` and expose Python bindings as well. The initial version will support the standard 200-point variant with 2 Jokers, while keeping the implementation configurable for other variants such as the 165-point version. 

I also plan to use `bridge.cc` and `hearts.cc` as references during implementation. 
The action space would include bids from 100 to 200 (step size 5) plus `Pass`, and card-play actions for all 54 cards. 

Before starting the implementation and opening a PR, I’d appreciate any guidance or requirements the maintainers might have.

**Comment by lanctot** [comment]
> Before starting the implementation and opening a PR, I’d appreciate any guidance or requirements the maintainers might have.

This all sounds good, please follow the steps in the [Adding a Game](https://github.com/google-deepmind/open_spiel/blob/master/docs/developer_guide.md#adding-a-game) part of the developer guide. Starting from bridge or spades is probably good (as hearts is not a 2-v-2 team game).

**Comment by edawite** [comment]
I'd like to contribute a C++ implementation of Quarto. I checked the current game registry and open pull requests and do not see an active implementation. I plan to include the deterministic perfect-information game logic, structured state/action support, focused rule tests, random simulations, and a regression playthrough. Please let me know if this overlaps with work already in progress.

**Comment by edawite** [comment]
Implementation is now available for review in #1557.

**Comment by jaisinha77777** [comment]
Implemented Mahjong (Riichi, four-player) — Python-only, opening a PR shortly. Heads up that it's a large one (~2,200 lines across game logic, scoring, and tests), so wanted to flag it here first per the guidance in #1511.

A couple of things worth knowing going in:

- Calling (Pon/Chi/Kan/Ron) is the classic problem for this game in OpenSpiel — it's a real-time interrupt mechanic that doesn't map onto simultaneous-move or turn-based dynamics directly (there's an old unresolved thread on this at #979). I implemented it as sequential, priority-ordered polling after each discard: Ron-eligible players are asked first (multi-ron supported), then the single Pon/Kan-eligible player if any, then Chi (only the next player). This reproduces standard priority without needing true simultaneity.
- Scoring covers the standard yaku set (~25 named hands) plus the common yakuman, fu/han calculation, dora/aka-dora/ura-dora, and the mangan-through-yakuman point table — split into its own module (mahjong_riichi_utils.py) so hand evaluation is unit-testable independent of game flow.
- Scope: one hand per episode (deal through win/draw), not a full multi-hand match with dealer rotation/score carryover — same pattern as how e.g. leduc_poker models one hand, not a session.

Happy to adjust based on feedback, especially on the calling-priority design since that's the part with no existing precedent in the codebase.

**Comment by Stochastic-Batman** [comment]
Hi @lanctot  and team!

I would like to volunteer to implement **Fox and Geese** (along with unit tests and Python bindings if needed). 

It's a classic medieval asymmetric, 2-player, perfect-information board game on a cross grid. Since it's in the public domain and has an interesting asymmetric dynamic (Fox captures/jumps, Geese trap/block), it should be a clean addition to the collection.

I'm ready to open a dedicated issue proposal / start working on the implementation if this sounds good!

**Comment by lanctot** [comment]
> Hi [@lanctot](https://github.com/lanctot) and team!
> 
> I would like to volunteer to implement **Fox and Geese** (along with unit tests and Python bindings if needed).

This? https://en.wikipedia.org/wiki/Fox_games

That sounds great, would be very welcome! Yes, please post an issue and encourage you to contribute a PR.

**Comment by Stochastic-Batman** [comment]
Thanks @lanctot!  

I'll be focusing specifically on **Fox and Geese** for now (excluding the other variants mentioned in that Wikipedia entry like Fox and Hounds or Halatafl). 

I'll open a dedicated issue for Fox and Geese now to track the proposal and PR!

**Comment by Stochastic-Batman** [comment]
Hello @lanctot and team!

I'd like to volunteer to implement **Can't Stop**: https://en.wikipedia.org/wiki/Can%27t_Stop_(board_game)

It's a push-your-luck dice game (Sid Sackson, 1980) - perfect-information with chance events, small state space and bust-or-bank decision each turn that I think makes it a nice complement to the existing dice games (Pig, Yacht) already in the collection.

Happy to open a dedicated issue proposal and start on the implementation if this sounds good!

**Comment by lanctot** [comment]
> Hello [@lanctot](https://github.com/lanctot) and team!
> 
> I'd like to volunteer to implement **Can't Stop**: https://en.wikipedia.org/wiki/Can%27t_Stop_(board_game)

I *love* Can't Stop. I've played probably over 500 games of it, on [Yucata](https://www.yucata.de/en) and every day on a mobile app. I've even used it in a [research paper](https://mlanctot.info/files/papers/nips11vrmcts.pdf). 😁

But it's currently published and used in online games, so it doesn't really count as part of the public domain. There's nothing that would stop you from submitting it, but we would need to consult our legal team in order to merging it into the master branch. See #770 for more discussion on this.

Now we've had a few successes in the past few years getting licenses to do this, and I'd be delighted if we got one for Can't Stop, so if you want to pursue this I'd whole-heartedly encourage and support it.

**Comment by Stochastic-Batman** [comment]
Thanks for the context on Can't Stop, and for the encouragement regardless - good to know the door isn't closed there!

For now I'd like to go with something with no licensing question at all: the **Royal Game of Ur** (https://en.wikipedia.org/wiki/Royal_Game_of_Ur), a ~4,600-year-old Mesopotamian race game with a public domain.

IMHO, it's a good fit for the collection for a few reasons: perfect information game with a four-sided tetrahedral dice, tactical decisions (which piece to advance, rosette squares granting an extra turn, capturing opponents by landing on them), and a small, tractable state space.

On rules: per Wikipedia, based on a cuneiform tablet and the shape of the gameboard, Irving Finkel (a British Museum curator) reconstructed the basic rules of how the game might have been played, with the object being to run the course of the board and bear all one's pieces off before one's opponent - https://en.wikipedia.org/wiki/Royal_Game_of_Ur#Basic_rules_proposed_by_Irving_Finkel.

Happy to open a dedicated issue and get started if this sounds good!

**Comment by jaisinha77777** [comment]
Opened the PR: #1590

**Comment by choosen23** [comment]
Hi @lanctot,

Two small things.

**Thread update:** it looks like two of the three games in the top post have since landed and the list hasn't been updated:
- Capture-the-Flag — `open_spiel/games/capture_the_flag/` (commit 7c02e7aa)
- Pursuit-evasion — `open_spiel/python/games/pursuit_evasion.py` (commit e8bd22c8)

So Mahjong appears to be the only remaining item.

**Volunteering:** I'd like to contribute a game, and I'd rather start with something small to get familiar with the API and the contribution process before proposing anything Mahjong-sized. I'd like to implement **[Konane](https://en.wikipedia.org/wiki/Kōnane)** (Hawaiian checkers) — a two-player, perfect-information, zero-sum game. It's traditional and public domain, so there are no copyright considerations. It isn't on the wanted list, so please tell me if you'd rather I didn't.

It's a close structural match to the existing Clobber implementation: same grid, adjacency-based capture, and the same "player with no legal move loses" terminal condition, so I'd use `clobber` as the template.

Planned design, in case you'd rather I did any of it differently:

- **C++**, under `open_spiel/games/konane/`, parameterized on `rows`/`columns` like Clobber. Default 8x8.
- **Multi-jump as a single compound action** — `(from_square, direction, num_jumps)`, giving `rows * cols * 4 * max_jumps` distinct actions (768 on 8x8). Konane multi-jumps must continue in the same direction and may stop after any jump, so the alternative would be the Checkers-style approach of holding the turn open across hops; a compound action avoids the extra mid-turn state and keeps the terminal condition to "no legal actions". Happy to switch if you prefer consistency with Checkers.
- **Standard opening**: Black removes one stone from a corner or the center, then White removes one stone orthogonally adjacent to that gap, then normal play. This adds two special-case opening turns but matches the published rules.
- Observation tensor: 3 planes (current player's stones / opponent's stones / empty), same as Clobber.
- Includes `konane_test.cc` with random simulation tests, a generated playthrough, `docs/games.md` entry, and cpplint-clean code.

Is this welcome, and does the design look reasonable to you? Happy to adjust before I start.

Thanks in advance!

**Comment by lanctot** [comment]
Thanks @choosen23 , thread updated to refer to new contributions. I left up the pursuit-evasion game because I'd also love to have one played on an arbitrary grid with different link types (a la Scotland Yard) where the recent contrubution is on a 2D space.

That sounds good, Kohane would be a welcome addition! (btw, I was just in Hawaii last month 🌴🏖️🍍🌞)  That design sounds quite consistent with the rest of the games and works. You may want to consider also supporting some of the [OpenSpiel 2.0 features](https://mlanctot.info/introducing_open_spiel_2.0/) (like state and observation structs, maybe also action structs), but it's also you can add after the first basic implementation.

**Comment by hshb7** [comment]
Hi, I would like to add Kayles as a new game. The implementation would support a configurable row of pins, with players removing either one pin or two adjacent pins and normal-play scoring. I plan to implement it in C++ with tests and a playthrough, following the existing simple games such as Nim. I searched the current games, issues, and PRs and could not find an existing implementation. Would this be a welcome contribution?

**Comment by lanctot** [comment]
> Hi, I would like to add Kayles as a new game. 

> Would this be a welcome contribution?

Yes, Kayles would be a welcome contribution,  and your plan sounds good. Thanks!

**Comment by lanctot** [comment]
> Yes, Kayles would be a welcome contribution, and your plan sounds good. Thanks!

Please note that there are about 6-7 games in the queue right now so it might take some time before we get to it.
